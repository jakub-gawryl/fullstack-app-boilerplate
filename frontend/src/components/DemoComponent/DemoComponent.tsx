import React from 'react';
import gql from 'graphql-tag';
import { useQuery } from 'react-apollo';
import styled from 'styled-components';

const query = gql`
{
  users {
    id,
    name,
    email
  }
}
`;

const StyledDemoComponent = styled.div`
  display: flex;
  min-height: 200px;
  padding: 10px 32px;
  line-height: 2em;
  max-width: 500px;
  border: 1px solid #aaa;
  border-radius: 20px;
  margin: 20px auto 0;

  .demo-component {
    &__user-list {
      list-style-position: inside;
    }
  }
`;

const DemoComponent: React.FC = () => {
  const { loading, error, data } = useQuery(query);

  return (
    <StyledDemoComponent>
      <div className="demo-component">
        <h3 className="demo-component__frontend-text">✅ Hello Frontend!</h3>
        <h3>✅ Hello GraphQL (users):</h3>
        {loading && <strong>⌛ Loading...</strong>}
        {error && <strong style={{color: 'red'}}>😔 GraphQL request failed: {error.message}</strong>}
        {data && (
          <ul className="demo-component__user-list">
            {data.users.map((user: any) => (
              <li key={user.id}>
                🙍‍♂️ {user.name} &lt;{user.email.toLowerCase()}&gt;
              </li>
            ))}
          </ul>
        )}
      </div>
    </StyledDemoComponent>
  );
};

export default DemoComponent;
