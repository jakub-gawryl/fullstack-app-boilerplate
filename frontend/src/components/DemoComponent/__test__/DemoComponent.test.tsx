import React from 'react';
import { render } from '@testing-library/react';
import DemoComponent from '../index';
import { ApolloProvider } from 'react-apollo';
import ApolloClient from 'apollo-boost';

const client = new ApolloClient({
  uri: '/graphql'
});

describe('<DemoComponent />', () => {
  it('renders correctly', () => {
    React.useState = jest.fn().mockReturnValue(['Hello from backend!', {}]);
    const target = render(
      <ApolloProvider client={client}>
        <DemoComponent />
      </ApolloProvider>
    );

    expect(target.asFragment()).toMatchSnapshot();
  });
});