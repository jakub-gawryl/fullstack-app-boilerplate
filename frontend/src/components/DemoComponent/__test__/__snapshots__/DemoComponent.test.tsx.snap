// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DemoComponent /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL ezufFl"
  >
    <div
      class="demo-component"
    >
      <h3
        class="demo-component__frontend-text"
      >
        ✅ Hello Frontend!
      </h3>
      <h3>
        ✅ Hello GraphQL (users):
      </h3>
      <strong>
        ⌛ Loading...
      </strong>
    </div>
  </div>
</DocumentFragment>
`;
